modules/core/shared/src/main/scala/lucuma/core/model/sequence/PlannedTime.scala

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.model.sequence
+
+import cats.Eq
+import cats.Monoid
+import cats.implicits.catsKernelOrderingForOrder
+import cats.syntax.eq.*
+import cats.syntax.foldable.*
+import cats.syntax.functor.*
+import cats.syntax.semigroup.*
+import lucuma.core.enums.ChargeClass
+import lucuma.core.enums.ObserveClass
+import lucuma.core.util.TimeSpan
+
+import scala.annotation.targetName
+import scala.collection.immutable.SortedMap
+
+opaque type PlannedTime = SortedMap[ChargeClass, TimeSpan]
+
+/**
+ * Planned time broken down by charge class.
+ */
+object PlannedTime {
+
+  val Zero: PlannedTime =
+    SortedMap.empty
+
+  def apply(charges: (ChargeClass, TimeSpan)*): PlannedTime =
+    SortedMap(charges*)
+
+  def from(it: IterableOnce[(ChargeClass, TimeSpan)]): PlannedTime =
+    SortedMap.from(it)
+
+  /**
+   * Creates a `PlannedTime` instance where the entirety of the step estimate
+   * is associated with the `ObserveClass`'s `ChargeClass`.
+   */
+  def fromStep(observeClass: ObserveClass, stepEstimate: StepEstimate): PlannedTime =
+    apply(observeClass.chargeClass -> stepEstimate.total)
+
+  extension (pt: PlannedTime) {
+
+    /**
+     * Gets the time charged for the given charge class (or TimeSpan.Zero if no
+     * charge is recorded).  Alias for `getOrZero`.
+     */
+    def apply(chargeClass: ChargeClass): TimeSpan =
+      getOrZero(chargeClass)
+
+    // N.B., if you refer to `apply` within this extensions object, it will
+    // find the Map apply :-/ For that reason I added `getOrZero`.
```

Comment:
Is this because it is an opaque type?

Reply:
Right, I would expect it to use the apply in the same extensions object but it doesn't seem to.